go/ephemeral/lib.go

```diff
@@ -388,6 +388,20 @@ func (e *EKLib) getOrCreateLatestTeamEKInner(ctx context.Context, teamID keybase
 	return teamEK, nil
 }
 
+// Try to get the TeamEK for the given `generation`. If this fails and the
+// `generation` is also the current maxGeneration, create a new teamEK.
+func (e *EKLib) GetMaybeCreateNewTeamEK(ctx context.Context, teamID keybase1.TeamID, generation keybase1.EkGeneration) (teamEK keybase1.TeamEk, err error) {
+	defer e.G().CTrace(ctx, "GetMaybeCreateNewTeamEK", func() error { return err })()
+
+	teamEK, err = e.G().GetTeamEKBoxStorage().Get(ctx, teamID, generation)
+	if err != nil {
```

Comment:
Should we check the error type here? Maybe we don't want to create new TeamEKs if you have some recurring disk error or something like that. You could be failing in a loop.

Reply:
good idea i'll use the custom errors we check for elsewhere to be safe-- to be clear though you wouldn't get stuck in a loop since once a new teamEK is created the failing one isn't the latest teamEK anymore, it's just some teamEK you can't access in the past